File: leetcode/searchInsert.js
var searchInsert = function(nums, target) {
    let begin = 0
    let end = nums.length - 1
    while (begin <= end) {
        let mid = begin + ((end-begin)/2)|0
        if (nums[mid] < target) begin = mid + 1
        else if (nums[mid] === target) return mid
        else end = mid - 1
    }
    return begin
}

File: leetcode/searchInsert.go
package leetcode

func searchInsert(nums []int, target int) int {
	low := 0
	high := len(nums)
	for low < high {
		mid := low + (high-low)/2
		if nums[mid] >= target {
			high = mid
		} else {
			low = mid + 1
		}
	}
	return low
}
